File: ProDy/Entropy.py
import time
from prody import calcShannonEntropy, parseMSA

msa_long = parseMSA("../data/PF00089_aligned.fasta")
msa_wide = parseMSA("../data/PF16957_aligned.fasta")

start = time.perf_counter()
calcShannonEntropy(msa_long)
elapsed = time.perf_counter() - start
print("[BENCH] Shannon entropy PF00089:", elapsed)

start = time.perf_counter()
calcShannonEntropy(msa_wide)
elapsed = time.perf_counter() - start
print("[BENCH] Shannon entropy PF16957:", elapsed)

File: Bio3D/Entropy.R
library(bio3d)
library(microbenchmark)

msa_long <- read.fasta("../data/PF00089_aligned.fasta")
msa_wide <- read.fasta("../data/PF16957_aligned.fasta")

# It's more feature than ProDy and MIToS implementations
long <- function() {
    entropy(msa_long)
}
wide <- function() {
    entropy(msa_wide)
}

bench <- microbenchmark(long(), times=1)

cat("[BENCH] Shannon entropy PF00089:", bench$time / 10^9, "\n", sep="")

bench <- microbenchmark(wide(), times=1)

cat("[BENCH] Shannon entropy PF16957:", bench$time / 10^9, "\n", sep="")
